### PR TITLE
WIP [RAINCATCH-1372] Update TSC and other packages

### PR DESCRIPTION
**Pull request by paolobueno**
## Motivation
Update typescript compiler version and other package versions.

Also fix documentation script bugs. (RAINCATCH-1399)

## Progress
- [x] Update tsc
- [x] Update other packages
- [x] Removing hoisting requirement from core
- [x] Create base tsconfig on repository root

### CODE REVIEW

File: client/datasync-client/tsconfig.json [hunk 1]
```diff
@@ -1,14 +1,7 @@
...
-    "noImplicitAny": false,
-    "experimentalDecorators": true,
-    "strictNullChecks": true,
-    "sourceMap": true
-  },
-  "include": [
-    "src/",
-    "test/"
-  ]
+    "extends": "../../tsconfig.json",
```
Comment (wtrocki) [inline client/datasync-client/tsconfig.json:2]: Great idea!

**Comment by wtrocki** [comment]
@feedhenry-raincatcher/rcdev  After this change `npm install` will be required (to instal new learna) and complete bootstrap. IT will also require new version of lerna.

**Comment by wtrocki** [comment]
@paolobueno - Did that compile to you? Getting multiple problems like this.
```
../../cloud/passportauth/src/auth/PassportAuth.ts (39,15): Argument of type 'RequestHandler' is not assignable to parameter of type 'PathParams'.\n  Type 'RequestHandler' is not assignable to type '(string | RegExp)[]'.\n    Property 'push' is missing in type 'RequestHandler'.

``

File: client/camera/package.json [hunk 2]
```diff
@@ -34,28 +34,28 @@
     "branches": 80
   },
   "devDependencies": {
-    "@types/bluebird": "^3.5.8",
-    "@types/chai": "^4.0.3",
+    "@types/bluebird": "^3.5.18",
```
Comment (wtrocki) [inline client/camera/package.json]: I think that using `^` is the main source of all our compilation problems. 
Not locking types brings some incompatibilities across the board. If we do not want to go back to shrinkwrap best to lock them - possibly separate ticket.

Comment (paolobueno) [reply client/camera/package.json]: It seems `npm-check-updates` has a --removeRange flag, I'll give it a try.

Comment (paolobueno) [reply client/camera/package.json]: Ideally we could start including the new npm 5+ package-locks in version control too.

**Comment by paolobueno** [comment]
I'm not getting any compilation errors, I did have some in travis that I fixed via removing a bad @types declaration, maybe a `lerna clean` would fix that one for you. Thanks for testing!

**Review by wtrocki — DISMISSED** [review]
Fixed by lerna clean

**Comment by wtrocki** [comment]
@paolobueno I have done lerna clean (it took a while to install all stuff again) but I can confirm that it works. I just forgot to update PR and approve this change.

**Comment by wtrocki** [comment]
@paolobueno e2c83cd updates only wfm module.

**Comment by wtrocki** [comment]
Changes tested with clean install.

**Review by wtrocki — APPROVED** [review]

**Pull request merged**